packages/grid/x-data-grid/src/utils/createSelector.ts

```diff
@@ -48,12 +48,99 @@ const missingInstanceIdWarning = buildWarning([
   'To fix, call it with `apiRef`, e.g. `mySelector(apiRef)`, or pass the instance ID explicitly, e.g `mySelector(state, apiRef.current.instanceId)`.',
 ]);
 
-export const createSelector: CreateSelectorFunction = (...args: any) => {
+function isAPIRef(value: any) {
+  return 'current' in value && 'instanceId' in value.current;
+}
+
+const DEFAULT_INSTANCE_ID = { id: 'default' };
+
+export const createSelector = ((
+  a: Function,
+  b?: Function,
```

Comment:
`b` is optional, but we throw an error if the `b` is not passed. Maybe:
```suggestion
  b: Function,
```